src/Core/Parser/TemplateParser.php

```diff
@@ -698,16 +698,16 @@ protected function recursiveArrayHandler($arrayText)
         if (preg_match_all(Patterns::$SPLIT_PATTERN_SHORTHANDSYNTAX_ARRAY_PARTS, $arrayText, $matches, PREG_SET_ORDER)) {
             foreach ($matches as $singleMatch) {
                 $arrayKey = $this->unquoteString($singleMatch['Key']);
-                if (!empty($singleMatch['VariableIdentifier'])) {
+                if (array_key_exists('Subarray', $singleMatch) && !empty($singleMatch['Subarray'])){
```

Comment:
I'm not sure this changes anything - and if it does, it merely hides the fact that the array was not detected correctly by PCRE. `$singleMatch['Subarray']` will still be set but will be null or an empty array, which is the core symptom of the problem. So this condition will still not trigger - and if it triggers with anything other than an array, it leads to new PHP errors (constructor of ArrayNode does not accept non-arrays).

Reply:
> it merely hides the fact that the array was not detected correctly by PCRE.
Yes, that is the point, see my comment above.

In our tests `VariableIdentifier` or `QuotedString` were set *additionally* to `Subarray` and that's why the wrong if/elseif-branch was executed. Changing the order did fix that. I agree that the check for the Subarray is not solid enough, an `is_array()` should be added.